ballerina/listener_utils.bal

```diff
@@ -102,6 +106,176 @@ isolated function addDefaultDirectives(__Schema schema) {
     }
 }
 
+isolated function getResponseFromMultipartPayload(Engine engine, Context context, http:Request request)
+    returns http:Response {
+    map<FileUpload> fileInfo = {};
+    map<json> pathMap = {};
+    map<json> variables = {};
+    json payload = ();
+    map<FileUpload|FileUpload[]> files = {};
+    mime:Entity[]|http:ClientError bodyParts = request.getBodyParts();
+    if bodyParts is mime:Entity[] {
+        foreach mime:Entity part in bodyParts {
+            mime:ContentDisposition contentDisposition = part.getContentDisposition();
+            if contentDisposition.name == MULTIPART_OPERATIONS {
+                json|mime:ParserError operation = part.getJson();
+                if operation is json {
+                    var variableValues = operation.variables;
+                    payload = operation;
+                    variableValues = variableValues is error ? () : variableValues;
+                    if variableValues is map<json> {
+                        variables = variableValues;
+                    } else {
+                        return createResponse("Invlaid Mulitpart Request", http:STATUS_BAD_REQUEST);
+                    }
+                } else {
+                    return createResponse(operation.message(), http:STATUS_BAD_REQUEST);
+                }
+            } else if contentDisposition.name == MULITPART_MAP {
+                json|mime:ParserError paths = part.getJson();
+                if paths is json {
+                    pathMap = <map<json>> paths;
+                } else {
+                    return createResponse(paths.message(), http:STATUS_BAD_REQUEST);
+                }
+            } else {
+                FileUpload|error handleFileFieldResult = handleFileField(part);
+                if handleFileFieldResult is FileUpload {
+                    fileInfo[part.getContentDisposition().name] = handleFileFieldResult;
+                } else {
+                    return createResponse(handleFileFieldResult.message(), http:STATUS_BAD_REQUEST);
+                }
+            }
+        }
+    } else {
+        return createResponse((<http:ClientError>bodyParts).message(), http:STATUS_BAD_REQUEST);
+    }
+
+    map<FileUpload|FileUpload[]>|string fileInfoResult = getFileUploadValues(fileInfo, pathMap, variables);
```

Comment:
Why do we need a `string` here? Can't we just return an `http:Response` directly from the `getFileUploadValues()` method?